include/revng/Model/QualifiedType.h

```diff
@@ -37,10 +37,34 @@ class model::QualifiedType : public model::generated::QualifiedType {
   std::optional<uint64_t> size() const debug_function;
   RecursiveCoroutine<std::optional<uint64_t>> size(VerifyHelper &VH) const;
 
+  /// \brief Checks if is a scalar type, without unwrapping typedefs
   bool isScalar() const;
+  /// \brief Checks if is a scalar type, unwrapping typedefs
+  bool isEventuallyScalar() const;
```

Comment:
* [ ] `\brief` is deprecated
* [ ] Honestly, the non-`Eventually` version sounds like a footgun. Let's be bold and not keep around unused, dangerous functions. This should also shrink this PR significantly.

Reply:
The non-`Eventually` versions are footguns unless they are exactly what you need. There's a couple of places in `revng-c` where they are exactly what we need. But well, we can probably hand-roll our step-one versions there. What do you think?

Reply:
Can you describe the `revng-c` use case?

Reply:
About `\brief`, what should I do? Switch to non-doxygen comments, or just remove `\brief` but keep the `///` comment?

Reply:
Just drop `\brief`, it's a [stupid feature](https://lists.llvm.org/pipermail/llvm-dev/2015-May/085152.html) that weshould stop using.
We still use doxygen though! `:)`

Reply:
The `revng-c` use case is when we're building pointer types in DLA, but now that I look more closely it's just assertions that a given type we've just built is a pointer at step one. Might as well use the `Eventually` version.

Reply:
The `\brief` thingy is gone. I'm happy to see it disappear. It was indeed stupid.